src/wallet/wallet.cpp

```diff
@@ -717,32 +717,54 @@ void CWallet::AddToSpends(const COutPoint& outpoint, const uint256& wtxid)
     SyncMetaData<COutPoint>(range);
 }
 
-void CWallet::AddToSpends(const uint256& nullifier, const uint256& wtxid)
+void CWallet::AddToSproutSpends(const uint256& nullifier, const uint256& wtxid)
 {
-    mapTxNullifiers.insert(make_pair(nullifier, wtxid));
+    mapTxSproutNullifiers.insert(make_pair(nullifier, wtxid));
 
     pair<TxNullifiers::iterator, TxNullifiers::iterator> range;
-    range = mapTxNullifiers.equal_range(nullifier);
+    range = mapTxSproutNullifiers.equal_range(nullifier);
     SyncMetaData<uint256>(range);
 }
 
-void CWallet::AddToSpends(const uint256& wtxid)
+void CWallet::AddToSproutSpends(const uint256& wtxid)
 {
     assert(mapWallet.count(wtxid));
     CWalletTx& thisTx = mapWallet[wtxid];
     if (thisTx.IsCoinBase()) // Coinbases don't spend anything!
         return;
 
     for (const CTxIn& txin : thisTx.vin) {
-        AddToSpends(txin.prevout, wtxid);
+        AddToSproutSpends(txin.prevout, wtxid);
     }
```

Comment:
It doesn't make sense to me to have transparent and Sprout spends handled in the same function, but Sapling spends handled in a different one.

I suggest:
* keep this as ``CWallet::AddToSpends(const uint256& wtxid)`` and make it handle Sapling spends as well (i.e. merge ``CWallet::AddToSaplingSpends(const uint256& wtxid)`` into it);
* rename ``CWallet::AddToSproutSpends(const COutPoint& outpoint, const uint256& wtxid)`` to ``AddToTransparentSpends``.